src/commands/Lastfm/Rank/AlbumRank.ts

```diff
@@ -41,47 +50,66 @@ export default class AlbumRank extends LastFMBaseCommand<typeof args> {
       { redirect: true }
     );
 
-    const topAlbums = await this.lastFMService.topAlbums(this.ctx, {
+    const rank = this.parsedArguments.rank;
+    const shouldSearchByRank = typeof rank === "number";
+
+    const topArguments = this.getTopArgs(rank, shouldSearchByRank);
+    let { albums } = await this.lastFMService.topAlbums(this.ctx, {
       username: requestable,
-      limit: 1000,
+      limit: topArguments.limit,
+      page: topArguments.page,
     });
 
-    const rank = topAlbums.albums.findIndex(
+    const originalAlbumCount = albums.length;
+
+    if (albums.length === 0)
+      throw new LogicError("You haven't scrobbled that many albums!");
```

Comment:
This should look like:
```ts
if (!albums.length) {
    throw new RankTooHighError("album");
}
```

or something, maybe you can think of a better name for the error. The error should go in the `errors.ts` file